gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Settings.java

```diff
@@ -345,4 +366,24 @@ public MessageSerializerV4<?> create() throws Exception {
             return serializer;
         }
     }
+
+    public static class AuthSettings {
+        /**
+         * Type of Auth to submit on requests that require authentication.
+         */
+        public String type = null;
```

Comment:
Nit: generally better to default Strings to empty string so you don't have to do a null check after just an isEmtpy() check.